question21/main.js

```diff
@@ -0,0 +1,159 @@
+    return json.data;
+  }
+  catch(error) {
+    console.error(error);
+  }
+  finally {
+    removeLoadImg();
+  }
+}
+
+const init = async() => {
+  const userData = await callApi();
+  if (userData){
+    renderTabaleElement(userData);
+    } else {
+    wrapper.textContent = "データの表示に失敗しました";
+    console.error('データの表示に失敗しました'); 
+  }
+}
+init();
+
+const userDataColumn = {
+  userId: "ID",
+  name: "名前",
+  gender: "性別",
+  age: "年齢"
+}
+
+const renderTabaleElement = (userData)  => {
+  const table = createElementWithClassName("table", "table");
+  const tbody = createElementWithClassName("tbody", "tbody");
+  table.id = "js-table";
+  tbody.id = "js-tbody";
+  const tableHeadValue = Object.values(userDataColumn);
+  wrapper.appendChild(table).appendChild(getTableHeadElement(tableHeadValue));
+  wrapper.appendChild(table).appendChild(tbody).appendChild(getTableRowFragment(userData));
+  sortButtonClickEvent(userData);
+}
+
+const getTableHeadElement = (tableHeadValue) => {
+  const thead = createElementWithClassName("thead", "thead");
+  const tr = createElementWithClassName("tr", "tr");
+  const fragmentThElement = document.createDocumentFragment();
+  for (const value of tableHeadValue) {
+    const th = createElementWithClassName("th", "th");
+    th.textContent = value;
+    value === 'ID' && th.appendChild(createSortButton());
+    fragmentThElement.appendChild(tr).appendChild(th);
+  }
+  thead.appendChild(fragmentThElement);
+  return thead;
+}
+
+const getTableRowFragment = (userData) => {
+  const tableRowfragment = document.createDocumentFragment();
+  for (let i = 0; i < userData.length; i++) {
+    const tr = createElementWithClassName("tr", "tr");
+    tableRowfragment.appendChild(tr).appendChild(getTableDataFragment(userData[i]));
+  }
+  return tableRowfragment;
+}
+
+const getTableDataFragment = (userData) => {
+  const tableDatafragment = document.createDocumentFragment();
+  Object.keys(userDataColumn).forEach((key) => {
+    const td = createElementWithClassName("td", "td");
+    td.textContent = userData[key];
+    tableDatafragment.appendChild(td);
+  });
+  return tableDatafragment;
+};
+
+const createSortButton = () => {
+  const sortButton = createElementWithClassName("button", "sortButton");
+  sortButton.dataset.status = "default";
+  sortButton.id = "js-sortbtn";
+  return sortButton;
+}
+
+const sortButtonClickEvent = (userData) => {
+  const tbody = document.querySelector('tbody');
+  const sortButton = document.getElementById('js-sortbtn');
+  sortButton.addEventListener('click', (e) => {
+    const nextStatus = getNextStatus(e.target);
+    sortButton.dataset.status = nextStatus;
+    const sortResultRows = sortFunc(e.target,userData);
+    while (tbody.firstChild) {
+      tbody.removeChild(tbody.firstChild);
+  }
+    sortResultRows.forEach((row) => {
+      tbody.appendChild(row);
+    });
+  });
+}
+
+const getNextStatus = (target) => {
+  switch (target.dataset.status) {
```

Comment:
IMO
I feel it would be easier to understand if it were named according to the switch statement
```suggestion
const switchSortStatus = (target) => {
  switch (target.dataset.status) {
```

Reply:
Thank you for your review! It is true. I will use your idea. 
fixed commit : https://github.com/csakyo/homework/pull/36/commits/4ea26032dc601575797f95d649bf7611819be7df